gensim/models/word2vec.py

```diff
@@ -1818,19 +1819,28 @@ def predict_output_word(self, context_words_list, topn=10):
 
         if not hasattr(self.wv, 'vectors') or not hasattr(self, 'syn1neg'):
             raise RuntimeError("Parameters required for predicting the output words not found.")
+        
+        # Retrieve indices if words were passed as input, otherwise keep the input indices
+        # Remark : out-of-vocabulary words are discarded.
+        word2_indices = []
+        max_index = self.wv.vectors.shape[0]
+        for w in context_words_list:
+            if w in self.wv:
+                word2_indices.append(self.wv.get_index(w))
+            elif isinstance(w, int) and (w < max_index):
+                word2_indices.append(w)
```

Comment:
I don't see this lines 1822-1831 approach as offering anything interesting over the 1-line list comprehension it's replacing. It's longer & less clear. The `w in self.wv` actually hides a call to `get_index`, so the intended optimization of not-calling-that-on-ints is premature & ineffective - this longer approach still already doing all the checks, & throwing an exception on negative or oversized ints, before then doing *yet more* redundant parameter-checking. 

The principles of "If It Ain't Broke Don't Fix It", "KISS", "Avoid Premature Optimization", etc all suggest: no reason to extend/change this method's parameter-checking at all to simply add the original tiny safe `np.sum()` improvement!

Reply:
All right, so the one-liner you're suggesting is
```python
word2_indices = [self.wv.get_index(w, default=-1) for w in context_words_list]
word2_indices = [i for i in word2_indices if i >= 0]
```
? Or did you have something else in mind ? The problem I find in this formulation is that it does not support negative indexing, as you suggested before. However in my opinion it's not a big deal and this formulation is ok to me too. If the user wants to use negative indexing, they can convert indices into positive ones before the call to `.predict_output_word()`.

Reply:
@gojomo sorry to bother you again on this PR, but could you please indicate me whether the solution I proposed in the last message suits you or if you were meaning something else ? I don't have a personal opinion and will abide by yours. Thanks in advance :)

Reply:
Sorry, I'd missed your question. I think the way it was is still cleanest:

    word2_indices = [self.wv.get_index(w) for w in context_words_list if w in self.wv]

Reply:
Oh, indeed. Mathis and I had not understood that `if w in self.wv` actually works to check both str and int values' validity as vector keys - I just reread the source code of `KeyedVectors` and figured it our. So, indeed, our additional checks, which were misguided by the impression that this check would not behave well with integers, are/were useless.

I'll push a fix right away.